lib/Backend/Lower.cpp

```diff
@@ -5710,9 +5710,6 @@ Lowerer::GenerateLdFldWithCachedType(IR::Instr * instrLdFld, bool* continueAsHel
 
     if (hasTypeCheckBailout)
     {
-        AssertMsg(PHASE_ON1(Js::ObjTypeSpecIsolatedFldOpsWithBailOutPhase) || !propertySymOpnd->IsTypeDead(),
```

Comment:
Can you leave the asserts but assert that either the type is dead or the bailout kind is equivalent?